Add docs for metric tables (#392)
santiment/academy: src/docs/santiment-queries/metric-tables/index.md

---
title: Metrics Datasets
author: Santiment Team
date: 2024-04-01
description: Learn about Santiment metrics datasets.
---

## Overview

This document introduces Santiment's precomputed metrics datasets.

You can filter all metric tables as:

```sql

> SHOW TABLES LIKE '%metric%'

Author: I dont think this is a issue, isn't it?

> SHOW TABLES LIKE '%metric%'

Author: Anyway I've changed it into `%metrics%` so metric_metadata is not shown

```

> Test in [Queries](https://queries.santiment.net/query/list-of-tables-390)

Reviewer: The example shows tables with `like '%price%'`

Author: Fixed

Usually, tables storing pre-computed data have a common set of columns:

- `dt` - A `DateTime` field storing the corresponding date and time.
- `asset_id` - An `UInt64` unique identifier for an asset. The data for that ID is stored in the `asset_metadata` table.
- `metric_id` - An `UInt64` unique identifier for a metric. The data for that ID is stored in the `metric_metadata` table.
- `value` - A `Float` column holding the metric's value for the given asset/metric pair.
- `computed_at` - A `DateTime` column storing the date and time when the given row was computed.

> Metrics are located in a different tables based on different parameters. All tables are similar to each other and described below.

Reviewer: ```suggestion
Metrics are located in different tables, depending on some different parameters. All tables are similar to each other and described below.
```
We should give examples. I was about to say that `intraday` contains intraday metrics and `daily` contains daily metrics, but there are a lot of daily metrics in intraday.

## Tables

---

### daily_metrics_v2

> Table `daily_metrics_v2` is the most basic table. It stores a signle metric value for each asset for each day. I.e. in that tables stored metrics that have exactly 1 value per day.

Reviewer: ```suggestion
The `daily_metrics_v2` table stores a single metric value for each asset for each day. I.e. in that tables stored metrics that have exactly 1 value per day.
```

For instance, you can retrieve ethereum balance of all centralized exchange addresses with the query:

```SQL
SELECT
    dt,
    value AS balance
FROM labeled_intraday_metrics FINAL
WHERE
    dt >= toDateTime('2024-01-01 00:00:00')
    AND label_id = dictGet('labels_by_fqn', 'label_id', 'santiment/centralized_exchange:v1')
    AND metric_id = dictGet('metrics_by_name', 'metric_id', 'combined_labeled_balance')
    AND asset_id = dictGet('assets_by_name', 'asset_id', 'ethereum')
    AND blockchain = 'ethereum'
ORDER BY dt DESC
LIMIT 5
```

Test in [Queries](https://queries.santiment.net/query/cex-balance-example-479)

---

### daily_label_based_metrics

Table `intraday_metrics` also has `label` and `owner` columns so that metrics stored in that table correspond to a particular label and owner.

> Note that this table uses old way to filter labels (plain label/owner instead of label_fqn).

This table could be used to retrieve [labelled exchange](/metrics/labeled-exchange) metrics under names: 

* `labelled_active_deposits`
* `labelled_deposit_transactions`
* `labelled_active_withdrawals`
* `labelled_withdrawal_transactions` 

> or NFT [trading volume](metrics/nft-trade-volume-usd) and [trades count](metrics/nft-trades-count) metrics.

Reviewer 2: Those links doesn't work. Maybe the metric has been renamed or removed?

Author: Should work once in master, Kate added these articles in a separate branch https://github.com/santiment/academy/pull/393
